lib/src/models/collection_reference.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:firedart/src/generated/google/firestore/v1/query.pb.dart';
+import 'package:firedart/src/generated/google/protobuf/wrappers.pb.dart';
+import 'package:firedart/src/generated/google/firestore/v1/document.pb.dart'
+    as fs;
+import 'package:firedart/src/models/document.dart';
+import 'package:firedart/src/models/document_reference.dart';
+import 'package:firedart/src/models/reference.dart';
+import 'package:firedart/src/repository/firestore_gateway.dart';
+import 'package:firedart/src/util/firestore_encoding.dart';
+
+class CollectionReference extends Reference {
```

Comment:
Please add documentation for all classes/methods - see how we do it in the monorepo project (Metrics)

Reply:
classes that you created for now - no need to update documentation for all classes in the project yet.